src/scripts/main.js

```diff
@@ -1,5 +1,35 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  if (e.target.closest('.wall')) {
+    const wallRect = wall.getBoundingClientRect();
+    const spiderWidth = spider.offsetWidth;
+    const spiderHeight = spider.offsetHeight;
+
+    const wallStyle = getComputedStyle(wall);
+    const wallBorderRght = parseInt(wallStyle.borderRightWidth);
+    const wallBorderBottom = parseInt(wallStyle.borderBottomWidth);
+
+    let xCoords = e.clientX - wallRect.left;
+    let yCoords = e.clientY - wallRect.top;
+
+    xCoords = Math.max(
+      0,
+      Math.min(
+        xCoords - spiderWidth / 2,
+        wallRect.width - spiderWidth - wallBorderRght * 2,
+      ),
+    );
+
+    yCoords = Math.max(
+      0,
+      Math.min(yCoords, wallRect.height - spiderHeight - wallBorderBottom * 2),
```

Comment:
Similar to the x-coordinate calculation, the `yCoords` calculation should use `wallBorderBottom` instead of multiplying it by 2. The border is applied only once at the bottom.